feat: 부스 상세화면 부스 이미지에 따라 StatubBar Icon, TopAppBar Navigation Icon 색상 동적 변경 처리

easyhooon opened the pull request:
<!--
✅ PR 제목 작성 가이드
형식: <라벨>: <작업 요약>
예: feat: 로그인 페이지 구현, fix: 버튼 클릭 버그 수정
-->

## 🔗 관련 이슈
<!-- 이 PR과 연결된 이슈 번호를 명시해주세요 (예: Close #123) -->
- Close https://github.com/Project-Unifest/unifest-android/issues/443

## 📙 작업 설명
<!-- 주요 수정 사항이나 개발 내용을 요약해주세요 -->
- 이미지의 luminance 값을 계산하여, 밝은 이미지의 경우 어두운 아이콘을, 어두운 이미지의 경우 밝은 아이콘을 적용

## 🧪 테스트 내역 (선택)
- [x] 주요 기능 정상 동작 확인
- [x] 브라우저/기기에서 동작 확인
- [x] 엣지 케이스 테스트 완료
- [x] 기존 기능 영향 없음

## 📸 스크린샷 또는 시연 영상 (선택)
<!-- UI 변경사항이 있다면 이미지나 GIF를 첨부해주세요 -->

Before
|어두운 이미지(라이트모드)|밝은 이미지(다크모드)|
|:--:|:--:|
|<img src="https://github.com/user-attachments/assets/d8825a53-5345-4285-9ee9-a4502aeb3807" width="300" />|<img src="https://github.com/user-attachments/assets/93927c53-1c9c-4e4d-ad90-220ac00d7fa2" width="300" />|

After
|어두운 이미지|밝은 이미지|
|:--:|:--:|
|<img src="https://github.com/user-attachments/assets/7d84fdb6-d9cf-4ac2-bb37-f4ddc2e40aaa" width="300" />|<img src="https://github.com/user-attachments/assets/542c8c70-2663-4189-827c-0aa71e43b080" width="300" />|

## 💬 추가 설명 or 리뷰 포인트 (선택)
<!-- 리뷰어가 중점적으로 봐야 할 부분이나 설명이 필요한 내용을 자유롭게 작성해주세요 -->
- 클로드는 신이다.


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- 신기능
  - 부스 썸네일 밝기 비동기 계산 도입(ImageLuminanceUtils)로 상태바·내비게이션 아이콘 색 자동 전환.
  - 일정 기반 부스 운영 여부 판단 유틸(isBoothCurrentlyRunning) 추가해 상세 설명에 '운영중' 상태 반영.

- 스타일
  - 뒤로가기 아이콘 디자인 정리(회색·흰색 버전 추가/조정) 및 기본 아이콘 색상 처리 개선(명시적 tint 제거).
  - 상단바(Search/QR/Booth 등) 아이콘 색상 일괄 조정.

- 작업
  - 이미지 로딩 라이브러리(coil.compose) 의존성 추가.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

easyhooon committed:
[feat] 부스 상세화면 이미지 luminance 계산 -> statusBar Icon, TopAppBar navigation Icon 색상 분기 처리 적용

easyhooon committed:
[chore] code style check

easyhooon committed:
[refactor] 토끼 리뷰 반영

- ImageLoader 싱글톤 방식으로 변경하여 호출
- 평균 밝기 계산의 분모와 실제 샘플 수를 일치

ikseong00 commented on core/common/src/main/kotlin/com/unifest/android/core/common/utils/ImageLuminanceUtils.kt:
```diff
@@ -0,0 +1,130 @@
+}
+
+@Composable
+fun rememberImageLuminance(imageUrl: String): LuminanceResult? {
+    val context = LocalContext.current
+    var luminanceResult by remember(imageUrl) { mutableStateOf<LuminanceResult?>(null) }
+
+    LaunchedEffect(imageUrl) {
+        if (imageUrl.isNotBlank()) {
+            luminanceResult = ImageLuminanceUtils.calculateLuminance(imageUrl, context)
+        }
+    }
```
key 값에 imageUrl 이 꼭 들어가야 하는 이유가 있을까요 ??
Url 특성상 바뀌지 않은 값인 것 같아 key 에서 빠져도 될 것 같습니다!

easyhooon replied:
저도 동의하는 바입니다. 지울께요!

그런데 관련해서 react에도 비슷한 useEffect라는 훅이 있는데 의존성배열(LaunchedEffect의 key와 대응)에 내부에서 사용하는 변수를 전부 명시하지 않으면 eslint의 rule에 걸리더라구여(exhaustive-deps)

왜 그런지 궁금해서 찾아봤는데, 정적 분석의 한계(실제로 변수가 바뀔 가능성이 없다는 것을 판단할수없음), 지금은 변하지 않더라도, 나중엔 코드가 수정되면서 변할수있는 값으로 바뀔 수 있음(안전빵으로), 일관성 등등...인데 요약하자면 '안전을 위해 조금 귀찮더라도 명시적으로 작성하자' 였던 것 같습니다.

ikseong00 commented on core/common/src/main/kotlin/com/unifest/android/core/common/utils/ImageLuminanceUtils.kt:
```diff
@@ -0,0 +1,130 @@
+                totalLuminance += luminance
+                sampleCount++
+            }
+        }
```
사소한 차이가 예상되지만,
비트맵 연산에서 꽤 많은 CPU 연산량이 요구될 것 같은데 `withContext(Dispatchers.Default)` 로 연산을 수행하는 건 어떨까요 ??

```suggestion
private suspend fun calculateBitmapLuminance(bitmap: Bitmap): LuminanceResult =
        withContext(Dispatchers.Default) { 
        // ... 
        LuminanceResult(averageLuminance, isDark)
}

```

easyhooon replied:
그렇네요 말씀하신것 처럼 Default가 더 적합한 것 같습니다. 바꿀께요!

ikseong00 reviewed (approved):
오.. 좋네요
비트맵 값을 하나하나 다 계산해서 평균치를 계산하는 방법
그냥 구현하려면 한참 걸릴 것 같은데.. 대클코

jisungbin commented on core/common/src/main/kotlin/com/unifest/android/core/common/utils/ImageLuminanceUtils.kt:
```diff
@@ -0,0 +1,130 @@
+@Composable
+fun rememberImageLuminance(imageUrl: String): LuminanceResult? {
+    val context = LocalContext.current
+    var luminanceResult by remember(imageUrl) { mutableStateOf<LuminanceResult?>(null) }
```
단순히 이 state 값을 계산하고자 LaunchedEffect이 사용된다면, 여기서 `produceState` 사용을 제안해 봅니당!

easyhooon replied:
아,, 또 remember + LaunchedEffect인데 produceState 안썼네여 [남의 코드 리뷰](https://github.com/YAPP-Github/Reed-Android/pull/175#discussion_r2311736531)할때는 보여도 내껀 안보이는군여 
반영하겠슴니둥! 감사합니다 ㅎㅎ

easyhooon committed:
[fix] 부스 상세화면 warning 텍스트 개행 적용

easyhooon committed:
[refactor] ImageLuminanceUtils 개선

Bitmap 연산 IO Dispatcher -> Default 로 변경
remember + LaunchedEffect -> produceState

easyhooon committed:
[refactor] 부스 상세 화면 현재 운영 중인 부스 판단 로직 Util로 이동

easyhooon commented:
> 비트맵 값을 하나하나 다 계산해서 평균치를 계산하는 방법 

아무래도 픽셀이 많으니 전부 계산하기엔 오래걸려서 샘플링(띄엄띄엄 픽셀 카운트) 적용해서 너무 계산이 오래걸리는걸 방지했슴니다!
대 클 코

jisungbin commented on core/common/src/main/kotlin/com/unifest/android/core/common/utils/ImageLuminanceUtils.kt:
```diff
@@ -0,0 +1,123 @@
+}
+
+@Composable
+fun rememberImageLuminance(imageUrl: String): LuminanceResult? {
```
이제 더 이상 `remember`가 사용되지 않는데, 그냥 이 함수가 사용되는 곳에 함수 로직을 직접 작성하는 건 어때요? 함수 로직이 엄청 간단해졌기도 하니깐요 😄.

easyhooon replied:
컴포즈  컨벤션상 그게 좋을 것 같네요!

easyhooon replied:
[2c62ea6](https://github.com/Project-Unifest/unifest-android/pull/444/commits/2c62ea6fa2001ca1d48e440971f365231034b920) 적용 완료 하였습니다 ㅎㅎ

jisungbin replied:
감사합니당!

easyhooon committed:
[refactor] rememberImageLuminance 함수 제거

내부에서 remember를 사용하지않음
직접 사용하는 곳에 적용하는게 더 간단(현재 한 곳에서만 사용)

easyhooon committed:
[chore] 토끼 리뷰 반영

luminance 계산 실패시 기존 전략(테마 기반 아이콘 색상) 적용

easyhooon commented on feature/booth-detail/src/main/kotlin/com/unifest/android/feature/booth_detail/BoothDetailScreen.kt:
```diff
@@ -104,10 +108,19 @@ internal fun BoothDetailRoute(
+    }.value
+    // 밝은 이미지일 때 어두운 아이콘 사용
+    val shouldUseDarkIcons = imageLuminance?.isDark == false
+
```
[29c915b](https://github.com/Project-Unifest/unifest-android/pull/444/commits/29c915b13ef1267c7946df1ad9f2625c7c0634d3) 반영 완료

jisungbin commented on feature/booth-detail/src/main/kotlin/com/unifest/android/feature/booth_detail/BoothDetailScreen.kt:
```diff
@@ -104,10 +108,19 @@ internal fun BoothDetailRoute(
 
-    DisposableEffect(systemUiController) {
+    // 이미지의 luminance를 계산하여 동적 색상 결정
+    val imageLuminance = produceState<LuminanceResult?>(initialValue = null, key1 = uiState.boothDetailInfo.thumbnail) {
```
```suggestion
    val imageLuminance by produceState<LuminanceResult?>(initialValue = null, key1 = uiState.boothDetailInfo.thumbnail) {
        if (uiState.boothDetailInfo.thumbnail.isNotBlank()) {
            value = ImageLuminanceUtils.calculateLuminance(uiState.boothDetailInfo.thumbnail, context)
        }
    }
``` 

짠
